google/cloud/apigee/registry/applications/v1alpha1/registry_conformance_report.proto

```diff
@@ -0,0 +1,91 @@
+syntax = "proto3";
+
+package google.cloud.apigee.registry.applications.v1alpha1;
+
+import "google/api/field_behavior.proto";
+import "google/cloud/apigee/registry/applications/v1alpha1/registry_styleguide.proto";
+import "google/cloud/apigee/registry/applications/v1alpha1/registry_lint.proto";
+
+option java_package = "com.google.cloud.apigee.registry.applications.v1alpha1";
+option java_multiple_files = true;
+option java_outer_classname = "RegistryConformanceReportProto";
+option go_package = "github.com/apigee/registry/rpc;rpc";
+
+// ConformanceReport describes how well an API Spec or a series of
+// API Specs conform to a specific API Style guide.
+message ConformanceReport {
+    // Resource name of the conformance report.
+    string name = 1 [
+        (google.api.field_behavior) = REQUIRED
+    ];
+
+    // Name of the style guide that this report pertains to.
+    string styleguide_name = 2 [
+        (google.api.field_behavior) = REQUIRED
+    ];
+
+    // A list of guideline report groups.
+    repeated GuidelineReportGroup guideline_report_groups = 3;
+}
+
+// GuidelineReport describes how well an API Spec or a series of
+// API Specs conform to a guideline within an API Style Guide.
+message GuidelineReport {
+    // Name of the guideline that this report pertains to.
+    string guideline_name = 1 [
+        (google.api.field_behavior) = REQUIRED
+    ];
+
+    // A list of rule report groups.
+    repeated RuleReportGroup rule_report_groups = 2;
+}
+
+// RuleReport provides information and feedback on a rule that 
+// a spec breaches within a guideline on an API Style Guide.
+message RuleReport {
+    // Name of the rule that the spec breaches.
+    string rule_name = 1 [
+        (google.api.field_behavior) = REQUIRED
+    ];
+
+    // Resource name of the spec that the rule was breached on.
+    string spec_name = 2 [
+        (google.api.field_behavior) = REQUIRED
+    ];
+
+    // A suggestion for resolving the problem.
+    string suggestion = 3;
+
+    // The location of the problem in the spec file.
+    LintLocation location = 4;
+}
+
+// GuidelineReportGroup is an abstraction that maps status
+// (PROPOSED, ACTIVE, DEPRECATED, DISABLED) to a list of 
+// guideline reports for guidelines of that status.
+message GuidelineReportGroup {
+    // Status of the guidelines in this report group.
+    Guideline.Status status = 1 [
+        (google.api.field_behavior) = REQUIRED
+    ];
+
+    // A list of guideline reports.
+    repeated GuidelineReport guideline_reports = 2 [
+        (google.api.field_behavior) = REQUIRED
+    ];
+}
+
+// RuleReportGroup is an abstraction that maps severity
+// (ERROR WARNING, INFO, HINT) to a list of rule reports for
```

Comment:
Question probably orthogonal to this PR: does it make sense to make severity an enumeration? What if the user wants more granularity than that?

Reply:
We've discussed adding a weight field (a float?) that could be used to compute scores based on report results. I think what we have here is good because it is simple and that the weight field (when added) might address what you have in mind.

Reply:
This is an interesting point. Let's consider the alternative approach, which is to make severity a string that the user specifies. I agree, doing something like this would definitely allow more flexibility in specifying granularity of severities. This is good because it can allow us to differentiate between, say, a syntax error and semantic error. However, here are some things to consider if we opt for such an approach:

- The conventions of typical linters usually define a set of severity levels. Spectral defines[ 4 severity levels](https://meta.stoplight.io/docs/spectral/ZG9jOjI1MTg5-custom-rulesets#severity) which were the inspiration for this proto definition. AIPs, which is what inspired the style guide definition makes use of "must", "should", and "may" (observed in this [guideline](https://google.aip.dev/140)), which correspond to "error", "warning", and "info/suggestion". If we want to follow the convention, then it makes sense for us to also define some set levels here.

- There would need to be some method of users ranking the importance of the rules they create. In our case, it is clear that ERROR > WARNING > INFO > HINT. However, how is this ranking system going to be preserved if we allow users to create arbitrary rules? There needs to be a mechanism that we'd have to think of for this.

- Can lead to inconsistency and disorganization in style guides. If I create two different style guides, and in one I refer to errors as "error" and in another style guide someone refers to it as "fail", it may create inconsistency and confusion to someone who is unaware that they're actually the same thing when we fomulate our conformance reports.

- Without specific severities to categorize lint results in, new users who are adding rules to an existing style guide may find it difficult to know the exact category to add errors in. This would perhaps require additional documentation for users to know the exact severity, whereas our categories in the enum are well documented and clear.

All in all, in light of us following the conventions of AIPs, I think it makes sense to leave this field as an enumeration of types for now, but this is a really good idea for an extension. Perhaps an alternative solution could be that we can create a way in the future to further break down errors into various error types, or warnings into various warning types, etc. (which the user provides). I will take note of this as an extension and consider this problem more for sure.

Reply:
These are great points. Thanks for thinking it through so thoroughly.